File: lib/storage/base_storage.dart
abstract class BaseStorage {
  //初始化
  Future<void> init();

  //保存数据
  void saveValue<T>(String key, T value);

  //加载数据
  T loadValue<T>(String key, {required T defaultValue});

  void removeAll();

  void remove(String key);
}

File: android/app/src/main/kotlin/com/molihuan/hlbmerge/storage/BaseStorage.kt
package com.molihuan.hlbmerge.storage

import android.content.Context

interface BaseStorage {
    //初始化
    fun init(context: Context)

    //保存数据
    fun <T> saveValue(key: String, value: T)

    //加载数据
    fun <T> loadValue(key: String, defaultValue: T): T

    fun removeAll()

    fun remove(key: String)
}